### PR TITLE
allow users to hide logs

### PR DESCRIPTION
**Pull request by cheahuychou**
Depends on #277

Allow users to hide logs from a service or dependency (i.e. exocom) by adding `silent: true` in `application.yml` like so:

```
dependencies:
  -  type: exocom
     version: 0.22.1
     silent: true

services:
  public:
    web:
      location: ./web-server
      silent: true
```

### CODE REVIEW

File: exo-run/src/cli.ls
Comment (hugoesthere) [inline exo-run/src/cli.ls]: It might be worth doing a single nested for loop to build `services` and `silenced-services`, makes it a bit easier to understand what's going on.

File: exo-setup/src/cli.ls
Comment (hugoesthere) [inline exo-setup/src/cli.ls]: Same here, might make more sense in a single nested loop.

**Comment by charlierudolph** [comment]
Please add some tests for this. I'm thinking some feature tests where we have a service that sends some output and verify that by default it gets printed and when silenced it does not.

File: exosphere-shared/src/logger.ls
Comment (hugoesthere) [inline exosphere-shared/src/logger.ls]: What about this? 
```
if !@silenced-roles.includes parsed-line.left
  ...
```

**Review by hugoesthere — APPROVED** [review]

**Comment by cheahuychou** [comment]
Reopening, the PR somehow got closed after #277 was merged.

**Comment by cheahuychou** [comment]
@charlierudolph  I have added the feature tests.

File: exo-run/features/run.feature
Comment (hugoesthere) [inline exo-run/features/run.feature]: `hiding logs from certain services or dependencies`

File: exo-run/features/run.feature
Comment (hugoesthere) [inline exo-run/features/run.feature]: Let's list `exocom#{some-version}` here as well, since we're testing that it doesn't print logs for exocom.

File: exo-run/features/run.feature
Comment (hugoesthere) [inline exo-run/features/run.feature]: Can we check that online text for the web service is printed? So we know the logger doesn't just hide all output text.

File: exo-run/features/run.feature
Comment (hugoesthere) [reply exo-run/features/run.feature]: You might also have to add a step before line 65 that checks that 'all services online' is printed. That way we're not checking for the absence of text before the application has been fully booted up.

File: exo-setup/features/setup.feature
Comment (hugoesthere) [inline exo-setup/features/setup.feature]: Didn't realize that output was being silenced for `exo-setup` as well. I think that can be a potential source of confusion for the user. For now we should always log output when setting up the application and only allow the option to hide output during `exo-run`.

But it's good to know that if we do want this feature in the future it's really easy to implement.

File: exo-setup/features/steps/steps-then.ls
Comment (hugoesthere) [inline exo-setup/features/steps/steps-then.ls]: See above, this change will be reverted.

File: exo-setup/src/cli.ls
Comment (hugoesthere) [inline exo-setup/src/cli.ls]: See above, this change will be reverted.

**Review by hugoesthere — APPROVED** [review]
Awesome! Good to go after comments are addressed and tests are green.

**Pull request closed without merging**
